src/endpoint/s3/s3_rest.js

```diff
@@ -241,8 +241,10 @@ async function authorize_request_policy(req) {
         return false;
     }());
 
+    const is_same_root_account_owner = req.object_sdk.nsfs_config_root && account.owner === owner_account.id; // NC NSFS case for IAM account
```

Comment:
1) > why is this check separated from all the other checks in is_owner? I would keep all ownership checks together.

I thought it would be more readable to write it in different conditions: we have 1)` is_owner` (root account case) and 2) `is_same_root_account_owner` IAM account.

2) > also I don't know why we check for nsfs_config_root at all - this seems unrelated - why wouldn't we use the same check in containers?

In the function, we already had a couple of implementations related to NC NSFS specifically (see above code). Since we didn't implement the owner in containerized this value will be undefined, and anyway the comparison can be false, I can change it.

Reply:
But we want the same to apply for containerized code as well, so the separation does not help.

Reply:
But currently containerized code doesn't have the IAM accounts, why should I combine it with a condition that is not relevant for it?

Reply:
The part `req.object_sdk.nsfs_config_root` was removed from the condition.